prusti-specs/src/specifications/preparser.rs

```diff
@@ -410,32 +391,29 @@ impl PrustiTokenStream {
 
         Some(Self { tokens, source_span: self.source_span })
     }
-
-    fn pop_one_nested_spec(self) -> Option<NestedSpec<Self>> {
-        // TODO: clean up the interface somehow ...
-        if self.tokens.is_empty() {
-            return None;
-        }
-        if self.tokens.len() != 2 {
-            panic!();
+    
+    fn pop_parenthesized_group(&mut self) -> syn::Result<Self> {
+        match self.tokens.pop_front() {
+            Some(PrustiToken::Group(_span, Delimiter::Parenthesis, box group)) => {
+                Ok(group) // TODO: need to clone()?
+            }
+            _ => Err(syn::parse::Error::new(self.source_span, "expected parenthesized group")),
         }
-        match [
-            &self.tokens[0],
-            &self.tokens[1],
-        ] {
-            [
-                PrustiToken::Token(TokenTree::Ident(ident)),
-                PrustiToken::Group(_span, Delimiter::Parenthesis, box group),
-            ] => {
-                if ident == "requires" {
-                    Some(NestedSpec::Requires(group.clone()))
-                } else if ident == "ensures" {
-                    Some(NestedSpec::Ensures(group.clone()))
-                } else {
-                    None
-                }
+    }
+    
+    fn pop_single_nested_spec(&mut self) -> syn::Result<NestedSpec<Self>> {
+        let first = self.tokens.pop_front().map_or_else(|| {
+            error(self.source_span, "expected nested spec")
+        }, Ok)?;
```

Comment:
A version of the `error` helper method which only returns the `syn::parse::Error` might be nice. Then this can be an `ok_or_else`, and the errors in `expr_bp` would be a bit nicer.